blog/2024-01-12-implement-async-storage-data-streaming-pytorch/index.md

```diff
@@ -0,0 +1,256 @@
+---
+title: "How to Implement Async Storage Data Streaming in Your PyTorch Project"
+description: Discover how to implement async storage data streaming in PyTorch projects. This blog provides a practical approach to setting up a custom PyTorch's IterableDataset that reads data from a time series database.
+authors: anthony
+tags: [AI, DataStreaming, PyTorch]
+slug: ai/datastreaming/pytorch/implement-async-storage-data-streaming-pytorch
+date: 2024-01-12
+image: ./img/pytorch-iterabledataset.webp
+---
+
+![PyTorch Training Diagram](./img/pytorch-iterabledataset.webp)
+<small>PyTorch training loop with data streaming from remote device</small>
+
+When training a model, we aim to process data in batches, shuffle data at each epoch to avoid over fitting, and leverage Python's `multiprocessing` for data fetching through multiple workers.
+
+The reason that we want to use multiple workers is that GPUs are capable of handling large amounts of data concurrently; however, the bottleneck often lies in the time-consuming task of loading this data into the system.
+
+Moreover, the challenge is even trickier when there is simply too much data to store the whole dataset on disk and we need to stream data from an async storage database such as [**ReductStore**](<https://www.reduct.store/>).
```

Comment:
async storage database - I would say data storage or just database. I'm also not sure if we should use async here. I use an asynchronous approach to develop ReductStore, but it is internal. We use the word async mostly when we talk about client SDKs.